### PR TITLE
Refetching of search filters in views, so that the latest value is used

### LINKED ISSUE #19480: "My Filters" query value does not update with changes made
## Expected Behavior

The query value in a search filter always reflects the query being used.

## Current Behavior

Changes to the search filter query are not reflected when the filter has been edited from a different widget, or edited in "My Search Filters".

## Steps to Reproduce (for bugs)

1. Create a dashboard widget, adding a search filter with the query `_exists_:timestamp` which is saved to "My Filters"
image
2. Add another widget to this dashboard which uses the same search filter from the previous step
3. Save the dashboard
4. Send messages to this Graylog instance and view results in the dashboard
5. Go to My Search Filters and edit the query, changing it to `_exists_:timestamp AND _exists_:does_not_exist`  and alter the title and description, then save the search filter
6. Return to the dashboard, there should be no results
7. Edit the filter on one of the widgets and observe that, even though the results in the widget have changed, the search filter query still displays the previous value `_exists_:timestamp` instead of the updated value `_exists_:timestamp AND _exists_:does_not_exist`, and that the title and description also do not reflect the changes made

This can also be repeated by editing the search filter from one widget, observing that the search results in both widgets are altered, but that the filter query value in the other widget does not reflect the updated query.

## Context

This is misleading when building content.

## Your Environment

* Graylog Version: 6.0.0
* Java Version: openjdk version "17.0.10" 2024-01-16
* OpenSearch Version: 2.13.0
* MongoDB Version:
* Operating System: Ubuntu 22.04.4 LTS
* Browser version: Firefox 126.0

### PR DESCRIPTION
**Pull request by luk-kaminski**
## Description

Prior to this PR, the state of the referenced search filters in a view could differ from the latest data saved in the global filters collection. This PR changes 
1. that we only save the ID to the referenced filter in the view in MongoDB
2. Refetch and add state of the referenced search filters on load of a view. 

fixes https://github.com/Graylog2/graylog2-server/issues/19480

## How Has This Been Tested?
Manually.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by luk-kaminski** [comment]
I've spawned a test instance (https://server-pr-21393.dev.torch.sh) and can still see problems that I have mentioned when reviewing previous solution -> https://github.com/Graylog2/graylog-plugin-enterprise/pull/15362#issuecomment-5476295896

EDIT: it only happens when you modify search filters and play with dashboard at the same time. If you re-enter the dashboard, things are refreshed. So I believe it is a good fix, even if some corner cases exist.

**Comment by luk-kaminski** [comment]
@janheise - github prevents me from approving... I guess that only you can do it...

**Comment by janheise** [comment]
> I've spawned a test instance (https://server-pr-21393.dev.torch.sh) and can still see problems that I have mentioned when reviewing previous solution -> [Graylog2/graylog-plugin-enterprise#15362 (comment)](https://github.com/Graylog2/graylog-plugin-enterprise/pull/15362#issuecomment-5476295896)
> 
> EDIT: it only happens when you modify search filters and play with dashboard at the same time. If you re-enter the dashboard, things are refreshed. So I believe it is a good fix, even if some corner cases exist.

Yeah, that's also what @linuspahl mentioned IMHO. Without a big change in state handling in the FE, those concurrent changes can not be addressed. And that would be a global/general change - not only related to this case.

File: graylog2-server/src/main/java/org/graylog/plugins/views/migrations/V20260908120000_StripReferencedSearchFilterAttributesInViews.java [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    private static final String REFERENCED_FILTER_TYPE = "referenced";
+    private static final Set<String> ALLOWED_FILTER_FIELDS = Set.of("type", "id");
+
+    private final MongoCollection<Document> viewsCollection;
+    private final ClusterConfigService clusterConfigService;
+
+    @Inject
+    public V20260908120000_StripReferencedSearchFilterAttributesInViews(MongoConnection mongoConnection,
+                                                                         ClusterConfigService clusterConfigService) {
+        this.viewsCollection = mongoConnection.getMongoDatabase().getCollection("views");
```
Comment (luk-kaminski) [inline graylog2-server/src/main/java/org/graylog/plugins/views/migrations/V20260908120000_StripReferencedSearchFilterAttributesInViews.java]: Do we limit the scope of the fix to views? I've noticed that the usage of search filters has grown in different part of the system, i.e. `SigmaRuleEntity`, `AggregationEventProcessorConfig` ...

Comment (janheise) [reply graylog2-server/src/main/java/org/graylog/plugins/views/migrations/V20260908120000_StripReferencedSearchFilterAttributesInViews.java]: as discussed, we move the migration to a separate PR as it's not really necessary to clean up existing views and we can also decide what we backport deparately

**Review by janheise — APPROVED** [review]
Approving this PR, @luk-kaminski and I tested/reviewed each others code but because he initially created the PR, I am the only other dev that can approve it.
